### PR TITLE
bitwise_and ONNX support

### PR DESCRIPTION
**Pull request by msftlincoln**
**Description**: Replace fallback with generated impl, add tests.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by msftlincoln** [comment]
This is the generated function:

// aten::bitwise_and.Tensor_out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
Tensor& bitwise_and_Tensor_out(
  const Tensor& self, 
  const Tensor& other, 
  // *, 
  Tensor& out) {
  ORT_LOG_FN(self, other, out);
  
  if (
    !IsSupportedType(self, {at::kBool}) || 
    !IsSupportedType(other, {at::kBool})) {
    return native::call_fallback_fn<
      &native::cpu_fallback,
      ATEN_OP(bitwise_and_Tensor_out)>::call(self, other, out);
  }
  auto& invoker = GetORTInvoker(self.device());
  
  // resize the output and then create output ort value to be updated.
  resize_output(invoker, dynamic_cast<ORTTensorImpl*>(out.unsafeGetTensorImpl()), self.sizes());
  auto ort_input_out = create_ort_value(invoker, out);
  
  auto ort_input_0_self = create_ort_value(invoker, self);
  auto ort_input_0_other = create_ort_value(invoker, other);
  
  std::vector<OrtValue> ort_outputs_0_And(1);
  ort_outputs_0_And[0] = ort_input_out;
  
  auto status = invoker.Invoke("And", {
    std::move(ort_input_0_self),
    std::move(ort_input_0_other),
  }, ort_outputs_0_And, nullptr);
  CHECK_STATUS(status);
  
  return out;
}

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 1]
```diff
@@ -272,6 +272,27 @@ def test_masked_select_broadcast(self):
...
         assert cpu_result.dim() == ort_result.dim()
 
+    def test_bitwise_and(self):
+        device = self.get_device()
+        cpu_a = torch.tensor([[0], [1], [1]], dtype=bool)
+        cpu_b = torch.tensor([[1], [0], [1]], dtype=bool)
+        ort_a = cpu_a.to(device)
+        ort_b = cpu_b.to(device)
+        cpu_result = torch.bitwise_and(cpu_a, cpu_b)
+        ort_result = torch.bitwise_and(ort_a, ort_b)
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:282]: add the out= param and verify those as well please.

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 2]
```diff
@@ -272,6 +272,27 @@ def test_masked_select_broadcast(self):
...
+
+    def test_bitwise_and_fallback(self):
+        device = self.get_device()
+        # use randint because bitwise_and is not supported on floats
+        cpu_a = torch.randint(200, (3, 4))
+        cpu_b = torch.randint(200, (3, 4))
+        ort_a = cpu_a.to(device)
+        ort_b = cpu_b.to(device)
+        cpu_result = torch.bitwise_and(cpu_a, cpu_b)
+        ort_result = torch.bitwise_and(ort_a, ort_b)
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:293]: i recently discovered OutputGrabber for checking output (used in orttraining/orttraining/eager/test/ort_eps_test.py). We don't need to use it hear but something to consider if we add output about falling back to verify behavior.

**Review by WilBrady — APPROVED** [review]

File: orttraining/orttraining/eager/test/ort_ops.py (hunk 1 above)
Comment (msftlincoln) [reply orttraining/orttraining/eager/test/ort_ops.py:282]: Verified

File: orttraining/orttraining/eager/test/ort_ops.py (hunk 2 above)
Comment (msftlincoln) [reply orttraining/orttraining/eager/test/ort_ops.py:293]: Cool, thanks!

**Pull request merged**